olive/azureml/azureml_client.py

```diff
@@ -41,7 +41,7 @@ class AzureMLClientConfig(ConfigBase):
     # but DefaultAzureCredential accept kwargs as parameters, it is hard to validate the config.
     # so we just provide a dict here and let the user to provide the correct config following the doc.
     default_auth_params: Optional[Dict[str, Any]] = Field(
-        None,
+        {},
```

Comment:
It is probably fine since pydantic doesn't appear to use the default value by reference. But it might be safer to use `Field(default_factory=dict, ...)` to be safe? In case there are unexpected issues with using a mutable object as default?

Reply:
Should `default_factory` in pydantic mainly be used for a callable in order to generate a dynamic default value?

Reply:
Looks like default_factory is not necessary. It's just that we have been avoiding using `{}` as default values (instead use config.x or {}) but we use other mutable objects as default values. 
So, it is probably fine to use `{}` as default.

Reply:
I think None as default value should be better than a default empty dict. 

Instead, whenever using the default_auth_params we need do validation

Reply:
I agree `None` as default looks better and follows what we do for other parameters in olive (where we use `None` instead and just do `param or {}` when needed).